src/helpers.rs

```diff
@@ -75,12 +74,15 @@ const UNIX_IO_ERROR_TABLE: &[(&str, std::io::ErrorKind)] = {
         ("ETIMEDOUT", TimedOut),
         ("ETXTBSY", ExecutableFileBusy),
         ("EXDEV", CrossesDevices),
-        // The following have two valid options. We have both for the forwards mapping; only the
-        // first one will be used for the backwards mapping.
+        // The following have multiple valid options. We have all of them for the forwards mapping;
+        // only the first one will be used for the backwards mapping.
         ("EPERM", PermissionDenied),
         ("EACCES", PermissionDenied),
         ("EWOULDBLOCK", WouldBlock),
         ("EAGAIN", WouldBlock),
+        ("ENOTSUP", Unsupported),
+        ("EOPNOTSUPP", Unsupported),
+        ("ENOSYS", Unsupported),
```

Comment:
This is a really unfortunate collision, but we are not the first to notice it. It was discussed on the libs PR that added `ErrorKind::Unsupported`: https://github.com/rust-lang/rust/pull/78880#issuecomment-777623233

Due to this degeneracy in the mapping, I think we should change our `strerror` mapping to not convert from libc constants to `ErrorKind`.

Reply:
Ah, right. The library decided to _not_ have this mapping farther down the thread: https://github.com/rust-lang/rust/pull/78880#issuecomment-778495108. I must have remembered this because of the `ENOSYS => ErrorKind::Unsupported` mapping that exists in the library.

Reply:
We're having the same conversation on about 4 different threads. I'm going to reply on the PR as a whole instead of on a diff.